src/Service/Database/Shell.php

```diff
@@ -0,0 +1,318 @@
+    }
+
+    /**
+     * @param OutputInterface $interface
+     */
+    public function setOutputAdapter(OutputInterface $interface)
+    {
+        $this->outputInterface = $interface;
+    }
+
+
+    /**
+     * Import the given backup file into the database.
+     *
+     * @param string $file Path to a file to import.
+     *
+     * @return void
+     * @throws \Symfony\Component\Process\Exception\RuntimeException
+     * @throws ServiceException
+     */
+    public function import($file)
+    {
+        $command = array_merge(
+            ['zcat ' . $file . ' | mysql'],
+            $this->getCredentialOptions(),
+            [$this->config->getDatabaseCredentials()->getName()]
+        );
+
+        $process = $this->createProcess($command);
+
+        $process->start();
+        $exitCode = $process->wait();
+
+        if ($exitCode !== 0) {
+            throw new ServiceException("There was an error exporting the database: " . $process->getErrorOutput());
+        }
+    }
+
+    /**
+     * Dump the database into a backup file.
+     *
+     * @param string $identifier An identifier for the dump file.
+     * @param string $strip_tables_patterns List of table patterns to dump with no data.
+     *
+     * @return string Path to the database dump.
+     * @throws \Symfony\Component\Process\Exception\LogicException
+     * @throws \Symfony\Component\Process\Exception\RuntimeException
+     */
+    public function dump($identifier, $strip_tables_patterns = '')
+    {
+        $strip_tables = implode(' ', $this->getStripTables($strip_tables_patterns));
+
+        $fileNamePrefix = $identifier . '-' . date('Y-m-d_His');
+
+        $structureOutputFile = $fileNamePrefix . '-structure.sql';
+        $dataOutputFile = $fileNamePrefix . '-data.sql';
+        $compressedFinalFile = $fileNamePrefix . '.sql.gz';
+
+        $databaseName = $this->config->getDatabaseCredentials()->getName();
+
+        /** @var Process[] $commands */
+        $commands = [];
+
+        if (trim($strip_tables) !== '') {
+            // Create the structure-only dump for tables that we don't want the data from.
+            $commands[] = $this->createDumpProcess([
+                '--no-data',
+                '--add-drop-table',
+                // @TODO Do we need to add --skip-triggers here?
+                // @TODO Do we need to add --routines here?
+                $databaseName,
+                $strip_tables,
+                "> $structureOutputFile"
+            ]);
+        } else {
+            // Empty structure file to make the rest of the code more consistent.
+            file_put_contents($structureOutputFile, '');
+        }
+
+        $dataDumpOptions = array_map(function ($table) use ($databaseName) {
+            return sprintf('--ignore-table=%s', escapeshellarg($databaseName . '.' . $table));
+        }, explode(' ', $strip_tables));
+
+        $commands[] = $this->createDumpProcess(array_merge($dataDumpOptions, [
```

Comment:
From @tonegolf71: We should add `--single-transaction --quick` to this (https://dev.mysql.com/doc/refman/5.7/en/mysqldump.html#option_mysqldump_single-transaction).